packages/components/src/components/CodeSnippet.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <v-popper placement="top" text="Click to copy" ref="popper">
+    <pre class="code-snippet" @click="copyToClipboard" ref="code"><slot /></pre>
+  </v-popper>
+</template>
+
+<script>
+import VPopper from './Popper.vue';
+
+export default {
+  name: 'CodeSnippet',
+  components: {
+    VPopper,
+  },
+  props: {
+    clipboardText: {
+      type: String,
+      required: false,
+      default: null,
+    },
+  },
+  methods: {
+    copyToClipboard() {
+      const text = this.clipboardText || this.$refs.code.innerText;
+      if (text === '' || !navigator || !navigator.clipboard) {
```

Comment:
shouldn't we check `clipboard.writeText` function is exists and open popper with error message for older browsers?
```suggestion
      if (text === '' || !navigator || !navigator.clipboard || !navigator.clipboard.writeText) {
        this.$refs.popper.flash('Can\'t copy to your clipboard!');
```

Reply:
Yeah, good point. Right now it'll just silently fail.

It's not too important now, though, because this is guaranteed be used in a modern browser (via vscode or IntelliJ). I've logged an issue to fail gracefully if the user is on an old browser.

https://github.com/applandinc/appmap-js/issues/309